moved/src/types/state.rs

```diff
@@ -159,45 +160,97 @@ impl From<Query> for StateMessage {
+            transactions: BlockTransactions::Hashes(
+                value
+                    .block
+                    .transactions
+                    .iter()
+                    .map(ExtendedTxEnvelope::compute_hash)
+                    .collect(),
+            ),
             header: alloy::rpc::types::Header {
                 hash: value.hash,
-                inner: alloy::consensus::Header {
-                    number: value.block.header.number,
-                    parent_hash: value.block.header.parent_hash,
-                    ommers_hash: value.block.header.ommers_hash,
-                    nonce: value.block.header.nonce.into(),
-                    base_fee_per_gas: Some(value.block.header.base_fee_per_gas.to_saturated_u64()),
-                    blob_gas_used: Some(value.block.header.blob_gas_used),
-                    excess_blob_gas: Some(value.block.header.excess_blob_gas),
-                    parent_beacon_block_root: Some(value.block.header.parent_beacon_block_root),
-                    logs_bloom: value.block.header.logs_bloom.into(),
-                    transactions_root: value.block.header.transactions_root,
-                    state_root: value.block.header.state_root,
-                    receipts_root: value.block.header.receipts_root,
-                    difficulty: value.block.header.difficulty,
-                    extra_data: value.block.header.extra_data,
-                    gas_limit: value.block.header.gas_limit,
-                    gas_used: value.block.header.gas_used,
-                    timestamp: value.block.header.timestamp,
-                    beneficiary: value.block.header.beneficiary,
-                    // TODO: review fields below
-                    mix_hash: Default::default(),
-                    requests_hash: None,
-                    withdrawals_root: None,
-                },
+                inner: value.block.header.into(),
                 // TODO: review fields below
                 total_difficulty: None,
                 size: None,
             },
-            transactions: BlockTransactions::Uncle,
             // TODO: review fields below
             uncles: Vec::new(),
             withdrawals: None,
+        })
+    }
+
+    pub fn from_block_with_transactions(value: ExtendedBlock) -> Self {
+        Self::new(alloy::rpc::types::Block {
+            transactions: BlockTransactions::Full(
+                value
+                    .block
+                    .transactions
+                    .into_iter()
+                    .filter_map(ExtendedTxEnvelope::into_canonical)
+                    .enumerate() // TODO: should index count deposited tx?
+                    .map(|(i, inner)| alloy::rpc::types::Transaction {
+                        block_hash: Some(value.hash),
+                        block_number: Some(value.block.header.number),
+                        transaction_index: Some(i as u64),
+                        effective_gas_price: None,
```

Comment:
Worth adding a TODO here referencing #160

Reply:
I'm not sure, the doc comment said it's deprecated.

Reply:
Yes, I see. The [PR where the field was introduced](https://github.com/alloy-rs/alloy/pull/1630) says that it is only used when the inner transaction type does not have its own gas price set. I that means we will still need to set this field in some circumstances because `TxEip1559` has `fn gas_price() -> None` in its implementation of `Transaction`.

Reply:
Good find.